src/main/java/com/itrex/java/lab/service/impl/ReporterServiceImpl.java

```diff
@@ -0,0 +1,35 @@
+package com.itrex.java.lab.service.impl;
+
+import com.itrex.java.lab.entity.dto.CustomerReportDTO;
+import com.itrex.java.lab.entity.dto.OfferReportDTO;
+import com.itrex.java.lab.repository.report.ReportRepository;
+import com.itrex.java.lab.repository.report.ReportRepositoryNativeSQL;
+import com.itrex.java.lab.service.ReporterService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDate;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class ReporterServiceImpl implements ReporterService {
+
+    private final ReportRepository repository;
+    private final ReportRepositoryNativeSQL repositoryNativeSQL;
+
+    @Override
+    @Transactional(readOnly = true)
+    public List<CustomerReportDTO> findAllCustomer(LocalDate firstStartContractDate, LocalDate endStartContractDate,
```

Comment:
In my opinion the opposite of 'first' is 'last' (and 'start-end')

Reply:
I also think that the name of the variables is too big. Start and end names will be sufficient and understandable

Reply:
Done.